packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -114,3 +118,52 @@ def test_ip_per_container_with_named_vip(dcos_api_session):
         with dcos_api_session.marathon.deploy_and_cleanup(proxy_app) as service_points:
             cmd = '/opt/mesosphere/bin/curl -s -f -m 5 http://foo.marathon.l4lb.thisdcos.directory:6000/ping'
             ensure_routable(cmd, service_points)()
+
+
+def geturl(tries, url):
+    rv = None
+    while tries > 0:
+        try:
+            rs = requests.get(url)
+            if rs.status_code == 200:
+                tries = 0
+                rv = rs.json()
+        except Exception as e:
+            logging.info("test_l4lb failed to connect to geturl {}: {}".format(url, e))
+        tries = tries - 1
+    return rv
+
+
+@pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
+def test_l4lb(cluster):
+    """Test l4lb is load balancing between all the backends
+    """
+    numapps = 5
+    numthreads = numapps * 4
+    tries = 100
+    apps = []
+    rvs = []
+    try:
+        for _ in range(0, numapps):
+            origin_app, origin_uuid = get_test_app()
+            origin_app['portDefinitions'][0]['labels'] = {'VIP_0': '/l4lbtest:5000'}
+            apps.append(origin_app)
+            sp = cluster.marathon.deploy_app(origin_app)
+            geturl(tries, 'http://{}:{}/ping'.format(sp[0].host, sp[0].port))
+            geturl(tries, 'http://l4lbtest.marathon.l4lb.thisdcos.directory:5000/ping')
+
+        def thread_request():
+            rvs.append(geturl(tries, 'http://l4lbtest.marathon.l4lb.thisdcos.directory:5000/test_uuid'))
+
+        threads = [threading.Thread(target=thread_request) for i in range(0, numthreads)]
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
+    except Exception as e:
+        logging.info("test_l4lb exception: {}".format(e))
+    for a in apps:
+        cluster.marathon.destroy_app(a['id'])
+    tids = [a['id'].split('-')[2] for a in apps]
+    rids = [r['test_uuid'] for r in rvs]
+    assert(set(tids) == set(rids))
```

Comment:
yea, distribution doesn't matter.
+1 on variable names!
why would `r` ever be None?